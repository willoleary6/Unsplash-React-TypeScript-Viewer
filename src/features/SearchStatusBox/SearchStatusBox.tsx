import React from "react";
interface SearchStatusBoxProps {
    messageToDisplay: string;
    isError: boolean;
}

export const SearchStatusBox = ({
    messageToDisplay,
    isError,
}: SearchStatusBoxProps): JSX.Element => {
    if (!isError) {
        return (
            <div className="container px-5 py-2 mx-auto lg:pt-12 lg:px-32 text-center bg-gray-300">
                <p>{messageToDisplay}</p>
            </div>
        );
    } else {
        return (
            <div
                className="bg-red-100 border border-red-400 text-red-700 px-4 py-3 rounded relative text-center"
                role="alert"
            >
                <span className="block sm:inline">{messageToDisplay}</span>
            </div>
        );
    }
};
