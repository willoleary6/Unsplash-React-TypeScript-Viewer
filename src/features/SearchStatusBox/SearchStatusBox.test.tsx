import React from "react";
import { SearchStatusBox } from "./SearchStatusBox";

import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({ adapter: new Adapter() });

describe("SearchStatusBox render testing", () => {
    it("renders panel and takes snapshot to track jsx element changes, will also test for exceptions in element code", () => {
        const wrapper = shallow(<SearchStatusBox messageToDisplay="test" isError={false} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders component in error mode and takes snapshot to track jsx element changes, will also test for exceptions in element code", () => {
        const wrapper = shallow(<SearchStatusBox messageToDisplay="test" isError={true} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
