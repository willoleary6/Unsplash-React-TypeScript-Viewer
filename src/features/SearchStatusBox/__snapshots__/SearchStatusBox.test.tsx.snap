// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchStatusBox render testing renders component in error mode and takes snapshot to track jsx element changes, will also test for exceptions in element code 1`] = `
<div
  className="bg-red-100 border border-red-400 text-red-700 px-4 py-3 rounded relative text-center"
  role="alert"
>
  <span
    className="block sm:inline"
  >
    test
  </span>
</div>
`;

exports[`SearchStatusBox render testing renders panel and takes snapshot to track jsx element changes, will also test for exceptions in element code 1`] = `
<div
  className="container px-5 py-2 mx-auto lg:pt-12 lg:px-32 text-center bg-gray-300"
>
  <p>
    test
  </p>
</div>
`;
